docs/content/intro/tutorial-preview-environments.md

```diff
@@ -0,0 +1,424 @@
+
+## Git Repositories
+
+We will use two GitHub repositories:
+
+- [brigade-tutorial-config](https://github.com/brigadecore/brigade-tutorial-config): containing orchestration responsible for managing Brigade projects and creating new environments.
+- [brigade-tutorial-app](https://github.com/brigadecore/brigade-tutorial-app): example microservice used to demonstrate release process.
+
+GitHub token will be used in this tutorial. To generate it follow [this article](https://help.github.com/en/articles/creating-a-personal-access-token-for-the-command-line).
+
+## Config Brigade Project
+
+Let's create new Brigade Project for our `brigade-tutorial-config` repository.
+
+```console
+$ brig project create --namespace brigade
+? Project Name brigadecore/brigade-tutorial-config
+? Full repository name github.com/brigadecore/brigade-tutorial-config
+? Clone URL https://github.com/brigadecore/brigade-tutorial-config
+```
+
+## Create Preview Environment Pipeline
+
+New environments will be created by executing brigade script via `brig` cli. 
+
+![Create Environment Pipeline](https://docs.brigade.sh/img/preview-environments-create.png)
+
+## Create Namespace
+
+Let's start by adding a new `brigade.js` script to the root of your repository. For now it will be responsible for creation of a new Kubernetes namespace.
+
+```js
+
+// brigade.js
+
+const { events } = require("@brigadecore/brigadier");
+const kubernetes = require("@kubernetes/client-node");
+
+const kubeConfig = new kubernetes.KubeConfig();
+kubeConfig.loadFromDefault();
+
+const k8sCoreClient = kubeConfig.makeApiClient(kubernetes.Core_v1Api);
+
+const protectedEnvironment = namespaceName => {
+  const protectedNamespaces = ["default", "kube-public", "kube-system", "brigade"];
+
+  if (protectedNamespaces.includes(namespaceName)) {
+    return true;
+  }
+  return false;
+};
+
+const createNamespace = async namespaceName => {
+  const existingNamespace = await k8sCoreClient.listNamespace(
+    true,
+    "",
+    `metadata.name=${namespaceName}`,
+  );
+
+  if (existingNamespace.body.items.length) {
+    console.log(`Namespace "${namespaceName}" already exists`);
+    return;
+  }
+
+  const namespace = new kubernetes.V1Namespace();
+  namespace.metadata = new kubernetes.V1ObjectMeta();
+  namespace.metadata.name = namespaceName;
+
+  await k8sCoreClient.createNamespace(namespace);
+};
+
+const provisionEnvironment = async (environmentName, projects) => {
+  await createNamespace(environmentName);
+};
+
+events.on("exec", event => {
+    const payload = JSON.parse(event.payload);
+    const { name } = payload;
+
+    if (!name) {
+      throw Error("Environment name must be specified");
+    }
+    if (protectedEnvironment(name)) {
+      throw Error(`Environment '${name}' is protected`);
+    }
+    provisionEnvironment(name, projects).catch(error => {
+      throw error;
+    });
+});
+
+```
+
+In the same directory add `payload.json` file with this content:
```

Comment:
Perhaps, `In the same directory, add a payload.json file with the following content:`